Tests/Document/DocumentTests.swift

```diff
@@ -0,0 +1,897 @@
+
+        target.update { root in
+            let data = root.data as? JSONArray
+            data?.append(Int64(3))
+            let one = data?.getElement(byIndex: 1) as? CRDTElement
+            try? data?.moveLast(id: one!.getID())
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"data":[0,2,3,1]}
+                           """)
+        }
+
+        XCTAssertEqual(target.debugDescription,
+                       """
+                       {"data":[0,2,3,1]}
+                       """)
+        XCTAssertEqual((target.getRoot().data as? JSONArray)?.length(), 4)
+    }
+
+    private var cancellables = Set<AnyCancellable>()
+
+    func test_change_paths_test_for_object() {
+        var paths = [String]()
+        let target = Document(key: "test-doc")
+
+        target.eventStream.sink { _ in
+
+        } receiveValue: { event in
+            XCTAssertEqual(event.type, .localChange)
+            XCTAssertEqual((event as? LocalChangeEvent)?.value[0].paths, ["$."])
+        }.store(in: &self.cancellables)
+
+        target.update { root in
+            root[""] = [:]
+            paths.append("$.")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{}}
+                           """)
+
+            let emptyKey = root[""] as? JSONObject
+            emptyKey!.obj = [:]
+            paths.append("$.obj")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{"obj":{}}}
+                           """)
+
+            let obj = emptyKey!.obj as? JSONObject
+            obj!.a = Int64(1)
+            paths.append("$.obj.a")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{"obj":{"a":1}}}
+                           """)
+
+            obj!.remove(key: "a")
+            paths.append("$.obj")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{"obj":{}}}
+                           """)
+
+            obj!["$.hello"] = Int64(1)
+            paths.append("$.obj.\\$\\.hello")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{"obj":{"$.hello":1}}}
+                           """)
+
+            obj!.remove(key: "$.hello")
+            paths.append("$.obj")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{"obj":{}}}
+                           """)
+
+            emptyKey!.remove(key: "obj")
+            paths.append("$")
+
+            XCTAssertEqual(root.debugDescription,
+                           """
+                           {"":{}}
+                           """)
+        }
+    }
+
+    func test_change_paths_test_for_array() {
```

Comment:
```change paths test for array``` of js-sdk has errors like below.
ERROR: 'YORKIE E:', AssertionError{message: 'expected [ '$.arr', '$.\\$\\$\\.\\.\\.hello' ] to deeply equal [ Array(6) ]', showDiff: true, actual: ['$.arr', '$.\$\$\.\.\.hello'], expected: ['$.arr', '$.arr.0', '$.arr.1', '$.arr', '$.\$\$\.\.\.hello', '$.\$\$\.\.\.hello.0'], operator: 'deepStrictEqual'}
```